src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -685,13 +778,13 @@ private async Task EnsureNuGetAndVsProjectAdapterCacheAsync()
             {
                 await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
 
-                IVsSolution ivsSolution = await _asyncVSSolution.GetValueAsync();
-                if (!_cacheInitialized && IsSolutionOpenFromVSSolution(ivsSolution))
+                if (!_cacheInitialized && _isSolutionOpen)
```

Comment:
This is potentially fragile isn't it? Is there a guarantee that InitializeAsync has run before this?

Reply:
This was a left-over from a previous attempt where I thought I could assume that `InitiaizeAsync` was always run before almost all methods are called, which didn't turn out to be true (for `SolutionDirectory`, `GetSolutionDirectory`, `IsSolutionopen`, `GetIsSolutionOpenAsync`).

And yes, looking at where `EnsureNuGetAndVsProjectAdapterCacheAsync` is called from, it's very difficult to figure out correctness. In theory, it should only be called when VsSolutionManager is already initialized. But looking at all the call locations, it's just too hard to be certain there isn't some edge case where that might not be true. So, it's changed now.